liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/UndeployAppMojo.java

```diff
@@ -31,41 +34,47 @@
  * Undeploy application from liberty server. If no parameters have been defined
  * the mojo will undeploy all applications from the server.
  *
- * @goal undeploy
+ * @Mojo( name = "undeploy" )
+ * 
  */
+@Mojo( name = "undeploy" )
+
 public class UndeployAppMojo extends BasicSupport {
     
     /**
      * A file name which points to a specific module's jar | war | ear | eba |
      * zip archive.
      *
-     * @parameter expression="${appArchive}"
-     * @optional
+     * @Parameter( property="appArchive" )
      */
+    @Parameter( property="appArchive" )
     protected String appArchive = null;
     
     /**
      * Maven coordinates of an application to undeploy. This is best listed as a
      * dependency, in which case the version can be omitted.
      *
-     * @parameter
+     * @Component( role = ArtifactItem.class )
      */
+    @Component( role = ArtifactItem.class )
     protected ArtifactItem appArtifact;
     
     /**
      * A set of patterns to include or exclude from the undeploy operation. If
      * appArchive or appArtifact has been defined then this parameter will be
      * ignored.
      *
-     * @parameter
+     * @Parameter( property="patternSet" )
      */
+    @Parameter( property="patternSet" )
```

Comment:
The documentation for @Component: Used to configure injection of Plexus components or Maven context components.  Are you sure we are using @Component correctly?  Can you run a test to make sure those complex type can be injected using @Component?